backend/src/graph-info/graph-info.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@nestjs/common';
+import { Neo4jService } from 'nest-neo4j/dist';
+import { parseNeo4jEntityInfo } from './parseNeo4jEntityInfo';
+import { EntityType } from '../shared/graph-information/EntityType';
+
+@Injectable()
+export class GraphInfoService {
+  constructor(private readonly neo4jService: Neo4jService) {}
+
+  /**
+   * Returns information about all nodes or edges of a graph
+   * @param which get graph info about nodes ('node') or relationships/edges ('rel')
+   */
+  async getEntityTypes(which: 'node' | 'rel'): Promise<EntityType[]> {
```

Comment:
Here the same applies as for the controller, but I would approve this, as this is hidden and not visible to other parts of the software (in particular the frontend).